src/client/sync_players.rs

```diff
@@ -0,0 +1,90 @@
+use crate::shared::{
+    components::{Grabbable, PlayerEntity},
+    messages::ServerMessages,
+    resource::Lobby,
+    sprites::{SpriteName, Sprites},
+};
+use bevy::prelude::*;
+use bevy_renet::netcode::NetcodeClientTransport;
+use bevy_renet::renet::*;
+pub struct ClientSyncPlayersPlug;
+
+impl Plugin for ClientSyncPlayersPlug {
+    fn build(&self, app: &mut App) {
+        app.add_systems(Update, client_sync_players);
+    }
+}
+
+pub fn client_sync_players(
+    mut commands: Commands,
+    mut client: ResMut<RenetClient>,
+    client_transport: Res<NetcodeClientTransport>,
+    mut lobby: ResMut<Lobby>,
+    sprites: Res<Sprites>,
+) {
+    while let Some(message) = client.receive_message(DefaultChannel::ReliableOrdered) {
+        let server_message = bincode::deserialize(&message).unwrap();
+        match server_message {
+            ServerMessages::PlayerConnected { client_id, ent_id } => {
+                let this_client_id = client_transport.client_id();
+                let player_entity_id = spawn_player_client(&mut commands, ent_id, &sprites);
+
+                if this_client_id == client_id {
+                    commands.entity(player_entity_id).insert(PlayerEntity);
+                }
+                lobby.players.insert(client_id, player_entity_id);
+            }
+
+            ServerMessages::PlayerDisconnected { id } => {
+                if let Some(player_entity) = lobby.players.remove(&id) {
+                    commands.entity(player_entity).despawn();
+                }
+            }
+            _ => {}
+        }
+    }
+    while let Some(message) = client.receive_message(DefaultChannel::Unreliable) {
+        match bincode::deserialize(&message) {
+            Ok(ServerMessages::SendPositions(players)) => {
+                for (player_id, transition) in players.iter() {
+                    if let Some(player_entity) = lobby.players.get(player_id) {
```

Comment:
Too many nests, maybe change to 
```suggestion
                    if let Some(player_entity) = lobby.players.get(player_id) else {
                         continue; // Or add logging for all errors
                    }
```

Reply:
Think about it in all the places we can do it

Reply:
make logs it is what we need I am sure